ml-agents/mlagents/trainers/policy/tf_policy.py

```diff
@@ -69,20 +69,26 @@ def __init__(
         self.update_dict: Dict[str, tf.Tensor] = {}
         self.sequence_length = 1
         self.seed = seed
-        self.brain = brain
+        self.behavior_spec = behavior_spec
 
-        self.act_size = brain.vector_action_space_size
-        self.vec_obs_size = brain.vector_observation_space_size
-        self.vis_obs_size = brain.number_visual_observations
+        self.act_size = (
```

Comment:
maybe out of scope, but I think act_size should be an int and we should have a branch_size for the multi-discrete case

Reply:
Yeah, I'm thinking we should have this mirror the behavior_spec for symmetry. Related thought: at some point we might need both continuous and discrete actions, so maybe we could further break it up into `discrete_act_size` and `cont_act_size`?

Reply:
spoiler warning: I'm going to push for implementing continuous + discrete actions soon.